jctools-experimental/src/main/java/org/jctools/channels/proxy/ProxyChannelFactory.java

```diff
@@ -286,8 +298,9 @@ private static void implementProcess(ClassVisitor classVisitor,
             int localIndexOfArrayReferenceBaseIndex = Integer.MIN_VALUE;
             for (Class<?> parameterType : method.getParameterTypes()) {
                 if (!parameterType.isPrimitive()) {
-                    // long referenceArrayIndex = this.consumerReferenceArrayIndex();
-                    consumerReferenceArrayIndex(methodVisitor, backendType);
+                    // long referenceArrayIndex = this.consumerReferenceArrayIndex(rOffset);
+                    consumerReferenceArrayIndex(methodVisitor, localIndexOfROffset, backendType);
+                    // TODO: Should reuse local indices or when an iFace grows too big we might have problems here
```

Comment:
While rereading this I noted that there's a risk of the locals table getting too big if we start proxying interfaces with lots of methods using references. e.g.

interface Alice {
  void f(Bob b);
  void g(Charlie c, David d);
}

Would result in 3 entries allocated in the locals table. Clearly this isn't necessary as the most needed is in the case of g(Charlie, David) which needs 2. This is due to the locals util being scoped for the entire method. Note an issue for small interfaces or within the context of this bug fix but I did add a comment to highlight it for the future.